edi_storage_oca/models/edi_backend.py

```diff
@@ -145,16 +145,24 @@ def _storage_create_record_if_missing(self, exchange_type, remote_file_name):
         return record.identifier
 
     def _storage_get_input_filenames(self, exchange_type):
+        full_input_dir_pending = exchange_type._get_full_exchange_type_path(
+            self.input_dir_pending
+        )
         if not exchange_type.exchange_filename_pattern:
-            # If there is not patter, return everything
-            return self.storage_id.list_files(self.input_dir_pending)
+            # If there is not pattern, return everything
+            files = self.storage_id.list_files(full_input_dir_pending)
+            filenames = []
+            for file in files:
+                if file.strip("/"):
+                    filenames.append(file)
+            return filenames
```

Comment:
```suggestion
            filenames = [x for x in self.storage_id.list_files(full_input_dir_pending) if x.strip("/")]
            return filenames
```
Wondering: when do you get `/` only?

Reply:
On this unit test:
https://github.com/OCA/edi/blob/15.0/edi_storage_oca/tests/test_edi_backend_storage.py#L206
The `list_files` function returns:
`['/', '/some-file.csv', '/another-file.csv']`